src/app_service/service/community/service.nim

```diff
@@ -1535,8 +1543,14 @@ QtObject:
           self.events.emit(SIGNAL_ACCEPT_REQUEST_TO_JOIN_FAILED, CommunityMemberArgs(communityId: communityId, pubKey: userKey, requestId: requestId))
         return
 
-      discard self.removeMembershipRequestFromCommunityAndGetMemberPubkey(communityId, requestId,
-        rpcResponseObj["response"]["result"]["communities"][0].toCommunityDto)
+      let updatedCommunity = rpcResponseObj["response"]["result"]["communities"][0].toCommunityDto
+
+      let newState = if updatedCommunity.isControlNode:
+          RequestToJoinType.Accepted
+        else:
+          RequestToJoinType.AcceptedPending
```

Comment:
Doesn't new response contain "requestsToJoin"? We could use that instead of duplicating the logic from status-go.

Reply:
I thought we would have it, but turns out the request to join isn't returned. I'll add it